.buildkite/dagster-buildkite/dagster_buildkite/utils.py

```diff
@@ -186,4 +188,17 @@ def get_changed_files():
         .strip()
         .split("\n")
     )
+    logging.info("Changed files:")
+    for path in paths:
+        logging.info(path)
     return [Path(path) for path in paths]
+
+
+def skip_if_no_python_changes():
+    if not is_feature_branch(os.getenv("BUILDKITE_BRANCH")):
+        return None
+
+    if not any(path.suffix == ".py" for path in get_changed_files()):
```

Comment:
we may consider memoizing get_changed_files if it gets called a lot and should be same for process life

Reply:
The good thing is it only runs on the initial step. So at most a few times. But yeah, memoizing is a good idea.